core/node/node_framework/src/implementations/layers/pk_signing_eth_client.rs

```diff
@@ -0,0 +1,54 @@
+use std::sync::Arc;
+
+use zksync_config::{ContractsConfig, ETHClientConfig, ETHSenderConfig};
+use zksync_eth_client::clients::PKSigningClient;
+
+use crate::{
+    implementations::resources::eth_interface::BoundEthInterfaceResource,
+    service::ServiceContext,
+    wiring_layer::{WiringError, WiringLayer},
+};
+
+#[derive(Debug)]
+pub struct PKSigningEthClientLayer {
+    eth_sender_config: ETHSenderConfig,
+    contracts_config: ContractsConfig,
+    eth_client_config: ETHClientConfig,
+}
+
+impl PKSigningEthClientLayer {
+    pub fn new(
+        eth_sender_config: ETHSenderConfig,
+        contracts_config: ContractsConfig,
+        eth_client_config: ETHClientConfig,
+    ) -> Self {
+        Self {
+            eth_sender_config,
+            contracts_config,
+            eth_client_config,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl WiringLayer for PKSigningEthClientLayer {
+    fn layer_name(&self) -> &'static str {
+        "pk_signing_eth_client_layer"
+    }
+
+    async fn wire(self: Box<Self>, mut context: ServiceContext<'_>) -> Result<(), WiringError> {
+        if self.eth_sender_config.sender.private_key().is_none() {
+            return Err(WiringError::Configuration(
+                "Private key is missing in ETHSenderConfig".to_string(),
+            ));
+        }
```

Comment:
It would be better for `PKSigningClient::from_config` to return an error in such cases instead of panicking.
It would probably even better to not have `PKSigningClient` to exist at all (logically it shouldn't know that there are some configs, it needs just a few variables).
But a quick attempt to refactor it has failed, so I decided not to do it in this PR and only catch the configuration error externally.